support/azureutil/azureutil.go

```diff
@@ -217,3 +226,64 @@ func VerifyResourceGroupLocationsMatch(ctx context.Context, hc *hyperv1.HostedCl
 
 	return nil
 }
+
+// GetAzureCredentialsFromSecret gets the Service Principal client ID, client secret, and tenant ID from the credentials
+// secret. This function will be modified a bit once the Microsoft sidecar containers support Managed Identity are
+// delivered (expected Oct 2024).
+func GetAzureCredentialsFromSecret(ctx context.Context, c client.Client, namespace, credsName string) (*corev1.Secret, error) {
```

Comment:
I could see this one split into 2. In general, any code that requires a client, we put in the main hcp controller. This could do the same with a *corev1.Secret as an argument.